posthog/queries/trends/sql.py

```diff
@@ -453,34 +440,35 @@
     -- for instance, month intervals which do not have a fixed number of seconds.
     periods AS (
         SELECT dateSub(
-            {{interval_expr}},
+            {interval_expr},
             number,
             dateTrunc(selected_period, toDateTime(%(date_to)s, %(timezone)s))
         ) AS start_of_period
         FROM numbers(
             dateDiff(
                 %(interval)s,
                 dateTrunc(%(interval)s, toDateTime(%(date_from)s, %(timezone)s)),
-                dateTrunc(%(interval)s, toDateTime(%(date_to)s, %(timezone)s) + INTERVAL 1 {{interval_expr}})
+                dateTrunc(%(interval)s, toDateTime(%(date_to)s, %(timezone)s) + INTERVAL 1 {interval_expr})
             )
         )
     )
-SELECT
-    groupArray(start_of_period) as date,
-    groupArray(counts) as data,
-    status
+SELECT groupArray(start_of_period) as date,
+        groupArray(counts) as data,
+        status
 FROM (
-    SELECT if(
+    SELECT
+        if(
```

Comment:
We can replace all of this zero padding by `WITH FILLS` (which I commented out in the metabase query), which also helps reduce subqueries, and removes the cross join altogether.

Might mean we need some special handling for statuses not present, but it's worth doing that in python vs. the query.

Reply:
Did you get to a working example query for that? I think the standard with fills doesn't work properly with intervals, so you have to do toRelativeMonthNum which was complicated.

In terms of performance, you're only doing those joins over the results, so a couple hundred rows at most. More keen to get this out than optimize to the nth degree

Reply:
Not blocking so okay to do in a follow up. Keen to hear more about what the issue was here 🤔 

This works as expected for me:

```
SELECT dateTrunc('week', toDateTime('2022-12-05 00:00:00', 'UTC')) as timestamp FROM numbers(1)
ORDER BY timestamp
WITH FILL
FROM dateTrunc('week', toDateTime('2022-11-03 00:00:00', 'UTC'))
TO dateTrunc('week', toDateTime('2023-03-03 00:00:00', 'UTC'))
STEP INTERVAL 1 WEEK
```